x-pack/legacy/plugins/apm/public/components/shared/DottedKeyValueTable/index.tsx

```diff
@@ -59,14 +59,18 @@ export function DottedKeyValueTable({
   data,
   parentKey,
   maxDepth,
-  tableProps = {}
+  tableProps = {},
+  skipPathify = false
 }: {
-  data: StringMap;
+  data: StringMap | PathifyResult;
   parentKey?: string;
   maxDepth?: number;
   tableProps?: EuiTableProps & TableHTMLAttributes<HTMLTableElement>;
+  skipPathify?: boolean;
 }) {
-  const pathified = pathify(data, { maxDepth, parentKey });
+  const pathified: PathifyResult = skipPathify
+    ? data
+    : pathify(data, { maxDepth, parentKey });
```

Comment:
Why do we need `skipPathify`? And not sure about `maxDepth` either

Reply:
Because the data was already pathified on MetadataTable component.

Reply:
Oh, okay. But perhaps this can be solved by creating the data structure up front, instead of inside the react component, as we talked about?

Reply:
yes, it will be.